### PR TITLE
Feature/recurrent donations

### PR DESCRIPTION
**Pull request by harshvitra**
### Adding Recurrent donations. 

- [x] Don't show recurrence if user is not logged in

- [x] For logged out users - Add hint - Login and continue to make a frequent donation.

- [x] From paymentSetup - Check isCollect flag inside the available gateways (paypal and stripe). If both are false for the project - Hide the recurrence option

- [x] Hide paypal if user selects anything except `Once` (as we don't support recurrence on paypal yet)

- [x] If user is logged in, grey out email in donation details page- regardless of type of donation - and send email as the account email. during donation request

- [x] Sync with the new version of the API

- [x] Hide Login text if recurrency is disabled

- [x] Add text near the cost to show the recurrency

- [x] allowRecurrency in tenant

### CODE REVIEW

File: src/features/donations/screens/TreeDonation.tsx [hunk 1]
```diff
@@ -312,6 +362,12 @@ function TreeDonation({
                 </div>
               )}
 
+            {!token && (
+              <div className={styles.loginHint} onClick={() => loginuser()}>
+                 Login to make a frequent donation.
```
Comment (norbertschuler) [inline src/features/donations/screens/TreeDonation.tsx:367]: Translation resource missing here.

File: src/features/donations/screens/TreeDonation.tsx [hunk 2]
```diff
@@ -286,7 +337,6 @@ function TreeDonation({
                   onKeyPress={() => setOpenTaxDeductionModal(true)}
                   role="button"
                   tabIndex={0}
-                  data-testid="taxDeductionId"
```
Comment (norbertschuler) [inline src/features/donations/screens/TreeDonation.tsx:289]: This is used by https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/106262e07ee0e156733bf9aacf9e8927c47bf122/__test__/TreeDonation.js#L18 (which is not yet an active test, but a demonstration).

**Review by norbertschuler — CHANGES_REQUESTED** [review]

**Comment by norbertschuler** [comment]
- [ ] On MacOS in latest Chrome the donation dialog is displayed not in the middle of the screen but on the lower part. But you can then scroll it up to the middle of the screen:

Bildschirmfoto 2021-01-18 um 11 01 02

**Comment by norbertschuler** [comment]
- [ ] Currently a recurring payment can not be finished due to a 202 response by the backend API (tested on develop backend) as following:

The request PUT https://app-development.plant-for-the-planet.org/app/donations/don_p2O50mQ3yuClgDPvBedObWTM return a status 202 with the response:
```json
{
  "id":"don_p2O50mQ3yuClgDPvBedObWTM",
  "status":"failed",
  "message":"payment method must already be attached to a customer"
}
```

Bildschirmfoto 2021-01-18 um 11 02 47

File: public/static/locales/en/donate.json [hunk 3]
```diff
@@ -13,7 +13,13 @@
...
   "youWillReceiveTaxDeduction": "You will receive a tax deduction receipt for ",
   "toProjectByOrganization": "To {{projectName}} by {{organizationName}}",
-  "fortreeCountTrees": "for {{treeCount}} Trees",
+  "fortreeCountTreesnone": "for {{treeCount}} Trees",
+  "fortreeCountTreesdaily": "for {{treeCount}} Trees every day",
+  "fortreeCountTreesweekly": "for {{treeCount}} Trees every week",
+  "fortreeCountTreesmonthly": "for {{treeCount}} Trees every month",
+  "fortreeCountTreesquaterly": "for {{treeCount}} Trees every three months",
+  "fortreeCountTreessemiyearly": "for {{treeCount}} Trees every six months",
+  "fortreeCountTreesyearly": "for {{treeCount}} Trees every year",
```
Comment (sagararyal) [inline public/static/locales/en/donate.json:22]: @rishsingh07 please make this one single item.

`{cost} {currency} for {treecount} Trees {frequency}`

Eg: 50 EUR for 50 Trees every month

if frequency = Once, 
`50 EUR for 50 Trees`

We only need Once, every month, and  every year.

Please remove other options for now.

Comment (norbertschuler) [reply public/static/locales/en/donate.json:22]: Appending `{frequency}` to this phrase works in German language - hopefully also in other languages - we probably have to explain which resources can be inserted for `{frequency}` when pushing this to LingoHub.

Comment (harshvitra) [reply public/static/locales/en/donate.json:22]: We just have one variable value which we are using to show the clickable options, we have to use the same variable to show to text as well, hence we cannot do it this way.

**Comment by norbertschuler** [comment]
- [ ] Testing recurring donations does not seems possible with the Stripe setup on the develop backend yet:

Request URL: https://app-development.plant-for-the-planet.org/app/donations/don_TQgJELtlLCylf5ebMEjDOYU6
Request Method: PUT
Status Code: 202 
Authorization: OAuth ***
tenant-key: ten_NxJq55pm
x-session-id: 1cc1aa10-4b6b-11eb-84e7-0b9fb52a46b8
Data: 
```JSON
{
  "paymentProviderRequest": {
    "account": "acct_1DYCMDD2OpW2f42N",
    "gateway": "stripe_pi",
    "source": {
      "id": "pm_1IDqPtGhHD5xN1Uq3XnNRmcM",
      "object": "payment_method"
    }
  }
}
```
Response: 
```JSON
{
  "id": "don_TQgJELtlLCylf5ebMEjDOYU6",
  "status": "failed",
  "errorCode": "resource_missing",
  "message": "No such PaymentMethod: 'pm_1IDqPtGhHD5xN1Uq3XnNRmcM'; OAuth key or Stripe-Account header was used but API request was provided with a platform-owned payment method ID. Please ensure that the provided payment method matches the specified account."
}
```

File: src/features/donations/screens/TreeDonation.tsx (hunk 2 above)
Comment (norbertschuler) [reply src/features/donations/screens/TreeDonation.tsx:289]: Can we add it again?

**Comment by norbertschuler** [comment]
- [ ] div-tags do not match:

```
error - ./src/features/donations/screens/PaymentDetails.tsx:363:6
Syntax error: Unexpected token, expected ","

  361 | 
  362 |         </div>
> 363 |       )
      |       ^
  364 |   ) : <></>;
  365 | }
  366 | 
```

File: src/features/donations/screens/TreeDonation.tsx (hunk 2 above)
Comment (harshvitra) [reply src/features/donations/screens/TreeDonation.tsx]: Done

**Comment by norbertschuler** [comment]
@sagararyal How are recurrent donations communicated to the user? 

- [ ] The _Thank you_ page does not display any information about the the fact the user has created a recurrent subscription. 
- [ ] The donation confirmation mail templates do not yet include any information about recurring donations. That could be added once the PR is released, but it would require the necessary changes in the backend.
- [ ] Can a user see or change his recurrent donations somewhere?

Bildschirmfoto 2021-02-06 um 13 46 55

File: public/tenants/planet/config.ts [hunk 4]
```diff
@@ -7,6 +7,7 @@ const config = {
   showUNEPLogo:true,
   showUNDecadeLogo:true,
   showRedeemHint:true,
+  allowRecurrecny:true,
```
[the remark this answers is not in the document]
Comment (norbertschuler) [reply public/tenants/planet/config.ts:10]: If you fix the typo, change it everywhere:
Bildschirmfoto 2021-02-16 um 15 00 08

**Comment by norbertschuler** [comment]
- [ ] I tried to do a recurrent payment with SEPA and got an error from Stripe:

```
Request URL: https://api.stripe.com/v1/payment_methods
Request Method: POST
Status Code: 400 
Payload:
 type: sepa_debit
 billing_details[name]: Norbert
 sepa_debit[iban]: NL25INGB8205778493
 guid: 51f471a6-6332-4a79-abe1-2a774166a0e8809277 
 muid: d196fc30-0d38-4821-b626-956b21c30935c291ea
 sid: da156703-c2a4-4853-930e-b7737a3333654302ad
 payment_user_agent: stripe.js/2fff16513; stripe-js-v3/2fff16513
 time_on_page: 106611
 referrer: http://localhost:3000/
 key: pk_test_zOadi2MBKX0gATOBvc3fzdRY00SLVj1YvJ
 _stripe_account: acct_1DYCMDD2OpW2f42N
Response:
{
  "error": {
    "code": "parameter_missing",
    "doc_url": "https://stripe.com/docs/error-codes/parameter-missing",
    "message": "Missing required param: billing_details[email].",
    "param": "billing_details[email]",
    "type": "invalid_request_error"
  }
}

```

**Comment by norbertschuler** [comment]
- [ ] Testing with a credit card resulted into a 500 error of the develop backend:

```
Request URL: https://app-development.plant-for-the-planet.org/app/donations/don_AATSjd2PN2mv2EViBuf5zDMN
Request Method: PUT
Status Code: 500 
Payload: {"paymentProviderRequest":{"account":"acct_1DYCMDD2OpW2f42N","gateway":"stripe_pi","source":{"id":"pm_1ILUJ4D2OpW2f42NWRujTU0U","object":"payment_method"}}}
Response: "Return value of App\\Common\\Email\\DatasetBuilder\\E08DonationConfirmationDatasetBuilder::getToEmail() must be of the type string, null returned\nfile: /app/src/Common/Email/DatasetBuilder/E08DonationConfirmationDatasetBuilder.php, line: 134"
```

**Comment by sagararyal** [comment]
@harshvitra can you rebase this to develop?

**Pull request closed without merging**